eval/eval_test.go

```diff
@@ -931,6 +931,20 @@ func TestMapAccidentalMutation(t *testing.T) {
 	resStr := res.Inspect()
 	expected := `{1:1,nil:"foo"}`
 	if resStr != expected {
-		t.Fatalf("wrong result, got %q expected %q", resStr, expected)
+		t.Errorf("wrong result, got %q expected %q", resStr, expected)
+	}
+}
+
+func TestSmallMapSorting(t *testing.T) {
+	inp := `m={2:"b"};n={1:"a"};m+n`
+	expected := `{1:"a",2:"b"}`
+	s := eval.NewState()
+	res, err := eval.EvalString(s, inp, false)
+	if err != nil {
+		t.Errorf("should not have errored: %v", err)
+	}
+	resStr := res.Inspect()
```

Comment:
Won't it panic on error?

I mean isn't res=nil on error?

What about this?


```suggestion
	if err != nil {
		t.Errorf("should not have errored: %v", err)
		return
	}
	resStr := res.Inspect()
```

Reply:
I don't think it does no, I think you get object.NULL but .. even if it does and the test crash, it's ok (agreed through that t.Fatalf() is what that'd be for

Reply:
like the test just above does, changed in #172

Reply:
Thanks